### PR TITLE
clarify pod_name behavior and change default key names to fit convention

### PR DESCRIPTION
**Pull request by sumo-drosiek**
docs(sourceprocessor): clarify pod_name behavior 
refactor(sourceprocessor)!: change default key names to fit k8s and this repo convention

change defaukt keys to be compatible with k8sprocessor

### CODE REVIEW

File: pkg/processor/sourceprocessor/README.md [hunk 1]
```diff
@@ -28,24 +28,30 @@ It has certain expectations on the label names used by `k8sprocessor` which migh
   an entry for `_SYSTEMD_UNIT`) then whenever the processed record contains
   `_HOSTNAME` attribute it will be added to the resulting record under `host` key.
 
-### Keys section (must match `k8sprocessor` config)
+### Keys section
 
-- `annotation_prefix` (default = "pod_annotation_"): prefix which allows to find given annotation; 
+The following keys must match resource attributes.
+In most of cases the keys should be the same like [k8sprocessor](../k8sprocessor/README.md#extract-section) config:
```
Comment (pmalek-sumo) [inline pkg/processor/sourceprocessor/README.md:34]:
```suggestion
In most cases the keys should be the same like in [k8sprocessor](../k8sprocessor/README.md#extract-section) config:
```

File: pkg/processor/sourceprocessor/README.md [hunk 2]
```diff
@@ -28,24 +28,30 @@ It has certain expectations on the label names used by `k8sprocessor` which migh
...
+
+- `annotation_prefix` (default = "k8s.pod.annotation."): prefix which allows to find given annotation;
```
Comment (pmalek-sumo) [inline pkg/processor/sourceprocessor/README.md:36]: Can we add code tags to surround the defaults with?

```suggestion
- `annotation_prefix` (default = `k8s.pod.annotation.`): prefix which allows to find given annotation;
```

File: pkg/processor/sourceprocessor/README.md [hunk 3]
```diff
@@ -28,24 +28,30 @@ It has certain expectations on the label names used by `k8sprocessor` which migh
...
-- `source_host_key` (default = "source_host"): attribute where source host is found
+- `namespace_key` (default = "k8s.namespace.name"): attribute where namespace name is found
+- `pod_key` (default = "k8s.pod.name"): attribute where pod full name is found
+- `container_key` (default = "k8s.container.name"): attribute where container name is found
+- `source_host_key` (default = "k8s.pod.hostname"): attribute where source host is found
+
+The following key is going to be created:
+
+- `pod_name_key` (default = "k8s.pod.pod_name"): attribute where name portion of the pod is stored
+during enrichment.
```
Comment (pmalek-sumo) [inline pkg/processor/sourceprocessor/README.md:48]: Can we provide an example for that?

Comment (sumo-drosiek) [reply pkg/processor/sourceprocessor/README.md]: even 3 examples :sweat:

**Pull request merged**